controllers/datadogagent/controller_reconcile_v2.go

```diff
@@ -95,6 +95,8 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 	newStatus := instance.Status.DeepCopy()
 
 	features, requiredComponents := feature.BuildFeatures(instance, reconcilerOptionsToFeatureOptions(&r.options, logger))
+	// update list of enabled features for metrics forwarder
+	r.updateMetricsForwardersFeatures(instance, features)
```

Comment:
Should we do this towards the end of the reconcile function? If something is enabled and reconciler starts failing the metrics reported would be inaccurate.

Reply:
that's a good point - Though I think we could actually keep the submission here to reflect that the feature is enabled, but later on, improve the code to have the value be `0` if for some reason it errors out (either alone or with a global error). WDYT?

Reply:
That's an interesting perspective, that way we would track intent of enabling feature and that's a reasonable thing to do. So, I could go either way.